modules/auxiliary/scanner/http/log4shell_scanner.rb

```diff
@@ -0,0 +1,193 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::TcpServer
+  include Msf::Auxiliary::Scanner
+
+  def initialize
+    super(
+      'Name' => 'Log4Shell HTTP Scanner',
+      'Description' => %q{
+        This module will scan an HTTP end point for the Log4Shell vulnerability by injecting a format message that will
+        trigger an LDAP connection to Metasploit. This module is a generic scanner and is only capable of identifying
+        instances that are vulnerable via one of the pre-determined HTTP request injection points. These points include
+        HTTP headers and the HTTP request path.
+      },
+      'Author' => [
+        'Spencer McIntyre'
+      ],
+      'References' => [
+        [ 'CVE', '2021-44228' ],
+        [ 'URL', 'https://attackerkb.com/topics/in9sPR2Bzt/cve-2021-44228-log4shell/rapid7-analysis' ]
+      ],
+      'DisclosureDate' => '2021-12-09',
+      'License' => MSF_LICENSE,
+      'DefaultOptions' => {
+        'SRVPORT' => 389
+      },
+      'Notes' => {
+        'Stability' => [CRASH_SAFE],
+        'SideEffects' => [IOC_IN_LOGS],
+        'AKA' => ['Log4Shell', 'LogJam'],
+        'Reliability' => []
+      }
+    )
+
+    register_options([
+      OptString.new('HTTP_METHOD', [ true, 'The HTTP method to use', 'GET' ]),
+      OptString.new('TARGETURI', [ true, 'The URI to scan', '/']),
+      OptPath.new('HEADERS_FILE', [
+        false, 'File containing headers to check',
+        File.join(Msf::Config.data_directory, 'exploits', 'CVE-2021-44228', 'http_headers.txt')
+      ]),
+      OptPath.new('URIS_FILE', [ false, 'File containing additional URIs to check' ]),
+      OptInt.new('TIMEOUT', [ true, 'Time to wait to receive LDAP connections', 5 ])
```

Comment:
This default value may need to be higher for the real world™

Reply:
Since we have `HTTP_TIMEOUT`, `SSH_TIMEOUT`, `RESPONSE_TIMEOUT`, etc. as a convention - potentially this could be `LDAP_TIMEOUT`?

```suggestion
      OptInt.new('LDAP_TIMEOUT', [ true, 'Time to wait to receive LDAP connections', 25 ])
```